### PR TITLE
Add proto docs

### PR DESCRIPTION
**Pull request by mattverse**
## Description

This PR adds proto docs in docs/core directory, which is auto-generated whenever we run `make proto-gen`. 
Also modified protocgen.sh file which is triggered by make proto-gen. 

As of now, the proto-docs.md includes all proto documentations except `gamm/pool-models/balancer` 
cref: https://github.com/bufbuild/buf/issues/99#issuecomment-655834414
(Also explains why we’re currently getting no errors when we do `proto-gen`)

We have duplicate symbols in https://github.com/osmosis-labs/osmosis/blob/main/proto/osmosis/gamm/pool-models/balancer/tx.proto#L9  and https://github.com/osmosis-labs/osmosis/blob/main/proto/osmosis/gamm/v1beta1/tx.proto#L9, and as a getaway ( we might need a permanent solution for this), restricted the maxdepth of proto docs generation, which excludes `gamm/pool-models/balancer` 

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Review by czarcas7ic — APPROVED** [review]
This is great, @daniel-farina knows how to set it up to track this from the docs site so I believe he can set that up after this gets merged.

At some point, we should take the time to add descriptions to all these protos. I think a majority of the osmosis specific ones are null.

File: scripts/protocgen.sh [hunk 1]
```diff
@@ -27,7 +27,20 @@ Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:. \
...
+
+# command to generate docs using protoc-gen-doc
+buf protoc \
+  -I "proto" \
+  -I "$cosmos_sdk_dir/third_party/proto" \
+  -I "$cosmos_sdk_dir/proto" \
+  --doc_out=./docs/core \
+  --doc_opt=./docs/protodoc-markdown.tmpl,proto-docs.md \
+  $(find "$(pwd)/proto" -maxdepth 4 -name '*.proto')
+go mod tidy
```
Comment (alexanderbez) [inline scripts/protocgen.sh:42]: Why do we tidy here?

In any case, we should ensure this has the 1.17 compat:

```
$ go mod tidy -compat=1.17
```

File: scripts/protocgen.sh [hunk 2]
```diff
@@ -27,7 +27,20 @@ Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:. \
   -I "$cosmos_sdk_dir/proto" \
   --grpc-gateway_out=logtostderr=true:. \
   $(find "${dir}" -maxdepth 1 -name '*.proto')
+  # command to generate docs using protoc-gen-doc
```
Comment (alexanderbez) [inline scripts/protocgen.sh:30]:
```suggestion
```

File: scripts/protocgen.sh (hunk 1 above)
Comment (mattverse) [reply scripts/protocgen.sh]: actually no need to tidy here!

File: scripts/protocgen.sh [hunk 3]
```diff
@@ -29,5 +29,15 @@ Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:. \
   $(find "${dir}" -maxdepth 1 -name '*.proto')
 done
 
+
```
Comment (alexanderbez) [inline scripts/protocgen.sh:32]:
```suggestion
```

**Review by alexanderbez — APPROVED** [review]
🌮🌮🌮

**Pull request merged**